### ISSUE TITLE
[Training]  TPR: 0.0000, FPR: 0.0000, PPV:0.0000, JS: 0.0000, DC: 0.0000, SE:0.0000, SP:0.0000

### ISSUE THREAD

**Issue by pangda72** [CN-zdy/MSU_Net#8]
Hi,CN-zdy
Thanks for sharing your code.when i run this repository,I got a problem.when i run" train.py",The evaluation results were all zero
#==================================================================================
Epoch [1/100], 
[Training]  TPR: 0.0000, FPR: 0.0000, PPV:0.0000, JS: 0.0000, DC: 0.0000, SE:0.0000, SP:0.0000
Valid......

[Validation]  TPR: 0.0000, FPR: 0.0000, PPV:0.0000, JS: 0.0000, DC: 0.0000, SE:0.0000, SP:0.0000
[Loss]: 36.5948  [lr]: 0.0010
Epoch [2/100], 
[Training]  TPR: 0.0000, FPR: 0.0000, PPV:0.0000, JS: 0.0000, DC: 0.0000, SE:0.0000, SP:0.0000
Valid......
#=================================================================================
I don't know how to deal with this problem.Could you please help me ?Looking forward to your reply!
Thanks  in advance.!
Best.

**Comment by CN-zdy** [comment]
Hi, Pangda.
Thank you for your interest in our work. There may be two reasons for this problem:
First, the Pytorch version is too high. You can try changing TP、FP... in evolution.py    ------->TP = ((SR==1).byte()+(GT==1).byte())==2 .FN = ((SR==0).byte()+(GT==1).byte())==2.
Second, there may be some problems with your groundtruth that you need to check carefully.

**Comment by pangda72** [comment]
Dear,CN-zdy
Thank you for your timely reply, I feel very happy.And,thank you for your advice,which could be a great help to me.
I succeeded in solving the problem just now by your advice.
thank you very much!Hope you have a good day!
Best.

**Comment by pangda72** [comment]
> Hi, Pangda.
> Thank you for your interest in our work. There may be two reasons for this problem:
> First, the Pytorch version is too high. You can try changing TP、FP... in evolution.py ------->TP = ((SR==1).byte()+(GT==1).byte())==2 .FN = ((SR==0).byte()+(GT==1).byte())==2.
> Second, there may be some problems with your groundtruth that you need to check carefully.

Dear,CN-zdy
Thank you for your timely reply, I feel very happy.And,thank you for your advice,which could be a great help to me.
I succeeded in solving the problem just now by your advice.
thank you very much!Hope you have a good day!
Best.
